src/Workspaces/Remote/Core/Services/RemoteWorkspace.cs

```diff
@@ -25,6 +25,12 @@ public RemoteWorkspace()
             Options = Options.WithChangedOption(CacheOptions.RecoverableTreeLengthThreshold, 0);
         }
 
+        // constructor for testing
```

Comment:
because of PrimaryWorkspace.Register set static variable, there used to be only one static RemoteWorkspace instance. except this new test .. I needed to create new workspace so that I can test AddSolution method. all the fun comming from static field.